src/aatoolbox/datasources/usgs/usgs_ndvi.py

```diff
@@ -0,0 +1,924 @@
+        ``geometries``, usually a geopandas
+        dataframes ``geometry`` feature. ``kwargs``
+        are passed on to ``aat.computer_raster_stats()``.
+        ``self.processed_file_suffix`` is used to define
+        the unique processed files, and should be changed
+        if the user wants to analyze across multiple files.
+
+        Parameters
+        ----------
+        gdf : geopandas.GeoDataFrame
+            GeoDataFrame with row per area for stats computation.
+            If ``pd.DataFrame`` is passed, geometry column must
+            have the name ``geometry``. Passed to
+            ``aat.compute_raster_stats()``.
+        feature_col : str
+            Column in ``gdf`` to use as row/feature identifier.
+            and dates. Passed to ``aat.compute_raster_stats()``.
+        clobber : bool, default = False
+            If True, overwrites existing processed dates. If
+            the new file matches the old file, dates will be
+            reprocessed and appended to the data frame. If
+            files do not much, the old file will be replaced.
+            If False, stats are only calculated for year-dekads
+            that have not already been calculated within the
+            file. However, if False and files do not match,
+            value error will be raised.
+        **kwargs
+            Additional keyword arguments passed to
+            ``aat.computer_raster_stats()``.
+
+        Returns
+        -------
+        Path
+            The processed path
+
+        Examples
+        --------
+        >>> from aatoolbox import create_country_config, \
+        ...  CodAB, UsgsNdviSmoothed
+        >>>
+        >>> # Retrieve admin 2 boundaries for Nepal
+        >>> country_config = create_country_config(iso3="npl")
+        >>> codab = CodAB(country_config=country_config)
+        >>> npl_admin2 = codab.load(admin_level=2)
+        >>> npl_admin1 = codab.load(admin_level=2)
+        >>>
+        >>> # setup NDVI
+        >>> UsgsNdviSmoothed(
+        ...     country_config=country_config,
+        ...     coverage_area="west_africa",
+        ...     start_date=[2020, 1],
+        ...     end_date=[2020, 3]
+        ... )
+        >>> UsgsNdviSmoothed.download()
+        >>> UsgsNdviSmoothed.process()
+        >>>
+        >>> # process for admin
+        >>> UsgsNdviSmoothed.processed_file_suffix = "adm1"
+        >>> UsgsNdviSmoothed.process()
+        """
+        processed_path = self._get_processed_path()
+
+        # get dates for processing
+        process_dates = _expand_dekads(
+            y1=self._start_year,
+            d1=self._start_dekad,
+            y2=self._end_year,
+            d2=self._end_dekad,
+        )
+
+        # check to see if file exists and remove
+        # if clobber or check dates already processed
+        # if not
+        if processed_path.is_file():
+            df = self.load()
+
+            # check that the processed file has the same analyzed
```

Comment:
So, even before, the `feature_col` was checked to make sure that the files matches with the `feature_col` for aggregation and the stats being calculated. However, I agree with y'alls comments that we should use `feature_col` for column identification, and have changed for that. I see the advantage of different files, just don't think that having a timestamp solves the problem of overwriting/metadata, since it doesn't help match up the current call with the processed file. I couldn't think of additional metadata to add, but having the `feature_col` seems pretty sufficient I think?

Reply:
hmmm yeah agree a lot is solved with adding the `feature_col`. Now I think the only case when it is overwritten is when using different stats, is that correct? 

One idea to prevent that is that we could have a separate file per stat? and then with loading combine them? Cause the risk I see now is that e.g. one person calls the function only computing `mean`. Someone else with only `sum`. And that would mean (ha pun) that suddenly the `mean` results disappear and would be hard to traceback why. What you think of that?

Reply:
That's another brilliant idea, and solves the rest of the issues. Will have a think through the cleanest way to implement this, and put something in to match.